### ISSUE TITLE
Opening existing repo with SSH private key password yields auth fail

### ISSUE THREAD

**Issue by dmusican** [dmusican/Elegit#540]
If you have a private repo that has been cloned via ssh authentication, when the private key is password protected, Elegit will spit authfail Transport Exceptions.

I'm taking this on because it is preventing me from testing Elegit on some of my own repos.

**Comment by dmusican** [comment]
There are multiple problems here that need resolving. One of them is that the RepositoryMonitor checks the remote every 5 seconds; if the password is bad, it keeps spewing errors. This needs to be thought through.

**Comment by dmusican** [comment]
Initial workaround, though this fix will not be complete: configure RepositoryMonitor so that if there is an auth fail, it stops doing remote checks, as long as that repo is open, period.

This needs to be repaired in a better way.

**Comment by dmusican** [comment]
I added a toggle to Preferences to at least disable remote checking if password is bad. 

A related problem preventing me from working on this where I am currently physically sitting at the public library, ssh is slow/blocked and results at the command line in just hanging when trying to make an ssh connection. When this happens, Elegit locks up. This needs to be threaded appropriately with a reasonable timeout, and appropriately tested. (I've filed this as a separate bug.)
